app.js

```diff
@@ -1,21 +1,29 @@
 const express = require('express');
+require("dotenv").config();
 
 const middleware = require('./utils/middlewares');
 const resources = require('./resources');
+const db = require('./database');
 
 const app = express();
 
 middleware(app);
 
-app.use(resources)
+app.use(resources);
 
-// catch 404 and forward to error handler
-app.use(function(req, res, next) {
-  next(createError(404));
-});
+(async () => {
+  try {
+    const conn = await db();
+    if (conn) {
+      console.log('connected to database');
+    }
+  } catch (e) {
+     console.log('Failed to connect to database');
+  }
+})();
```

Comment:
This should be moved to the model folder `model/index.js` and wrap in a function may be called `connectDB` that we could be called in our entry point file.

Reply:
I am not sure why this is necessary, because i could easily just call db() in app.js and that will suffice

Reply:
Are you sure this is a required step ?

Reply:
One of the reasons why I ask for the changes is because you use `IIFE` which I think is not necessary and the `conn` variable is not used anywhere in this file and the block doesn't explicitly say what it does

Reply:
Thinking of big-picture what if the first connection to the database failed due to network issue and you are asked to write a code that can make the request again how will that affect the length of the code block

Reply:
Remember `simple` is better than `complex`

Reply:
I get you, i only used an IIFE because i wanted to use an async and had to wrap in an async function. Anyways, but you can always call db() but nonetheless i will add the connection. Thanks for the feedback